bin/generate_practice_exercise.sh

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env bash
+
+# shellcheck source=/dev/null
+source ./bin/generator-utils/utils.sh
+source ./bin/generator-utils/prompts.sh
+source ./bin/generator-utils/templates.sh
+
+# Exit if anything fails.
+set -euo pipefail
+
+# If argument not provided, print usage and exit
+if [ $# -ne 1 ]; then
+    echo "Usage: bin/generate_practice_exercise.sh <exercise-slug>"
+    exit 1
+fi
+
+# Check if sed is gnu-sed
+if ! sed --version | grep -q "GNU sed"; then
+    echo "GNU sed is required. Please install it and make sure it's in your PATH."
+    exit 1
+fi
+
+# Check if jq and curl are installed
+command -v jq >/dev/null 2>&1 || {
+    echo >&2 "jq is required but not installed. Aborting."
+    exit 1
+}
+command -v curl >/dev/null 2>&1 || {
+    echo >&2 "curl is required but not installed. Aborting."
+    exit 1
+}
+
+# Check if exercise exists in configlet info
+check_exercise_existence "$1"
+
+# ==================================================
+
+SLUG="$1"
+UNDERSCORED_SLUG=$(dash_to_underscore "$SLUG")
+EXERCISE_DIR="exercises/practice/${SLUG}"
+AUTHOR_NAME=$(get_author_name)
+message "info" "You entered: $AUTHOR_NAME. You can edit this later in .meta/config.json.authors"
+EXERCISE_NAME=$(get_exercise_name "$SLUG")
+message "info" "You entered: $EXERCISE_NAME. You can edit this later in config.json"
+EXERCISE_DIFFICULTY=$(get_exercise_difficulty)
+message "info" "EXERCISE_DIFFICULTY is set to $EXERCISE_DIFFICULTY. You can edit this later in config.json"
+
+echo "Creating Rust files"
+cargo new --lib "$EXERCISE_DIR" -q
+mkdir -p "$EXERCISE_DIR"/tests
+touch "${EXERCISE_DIR}/tests/${SLUG}.rs"
+
+create_test_file_template "$EXERCISE_DIR" "$SLUG"
+create_lib_rs_template "$EXERCISE_DIR" "$SLUG"
+create_example_rs_template "$EXERCISE_DIR"
```

Comment:
It might be nice to have the example file also have the function definition to implement, like the `lib.rb`. 

So instead of:

```rust
// Create a solution that passes all the tests
```

You'd have:

```rust
fn food_chain() {
   // TODO: Create a solution that passes all the tests
   unimplemented!("implement food-chain exercise")
}

```